pkg/fqdn/name_manager.go

```diff
@@ -95,41 +107,35 @@ func (n *NameManager) GetModel() *models.NameManager {
 // Selector will be returned as CIDR identities, as other DNS Names which have
 // already been resolved may match this FQDNSelector.
 func (n *NameManager) RegisterForIdentityUpdates(selector api.FQDNSelector) []identity.NumericIdentity {
```

Comment:
Could you rename this to `RegisterForIdentityUpdatesLocked`  to make it more obvious that it doesn't lock for the caller (also Unregister).